avl_tree/avl.py

```diff
@@ -0,0 +1,141 @@
+class node:
+    def __init__(self, data, parent):
+        self.data = data
+        self.left = None
+        self.right = None
+        self.parent = parent
+        self.height = 0
+
+    def balance(self):
+        left = self.left.height if self.left else -1
+        right = self.right.height if self.right else -1
+        return left - right
+
+
+class tree:
+    def __init__(self, root):
+        self.head = node(root, None)
+        self.count = 0
+
+    def insert(self, root, key):
+        if root.data < key:
+            if root.right is not None:
+                self.insert(root.right, key)
+            else:
+                root.right = node(key, root)
+                root.right.parent = root
+                self.count += 1
+                self.updatebal(root.right)
+        else:
+            if root.left is not None:
+                self.insert(root.left, key)
+            else:
+                root.left = node(key, root)
+                root.left.parent = root
+                self.count += 1
+                self.updatebal(root.right)
+
+    def add(self, key):
+        root = self.head
+        self.insert(root, key)
+
+    def search(self, root, value):
+        if root.data == value:
+            return root
+        elif root.left is not None and root.right not None:
+            self.search(root.left, value)
+            self.search(root.right, value)
+        elif root.left is not None:
+            self.search(root.left, value)
+        elif root.right is not None:
+            self.search(root.right, value)
+
+    # def height(self,root): #root is a node
+    #     if root.left!=None and root.right!=None:
+    #         return 1+ max(self.height(root.left),self.height(root.right))
+    #     elif root.left!=None:
+    #         return 1+ self.height(root.left)
+    #     elif root.right!=None:
+    #         return 1+ self.height(root.right)
+    #     else:
+    #         return 1
+    #
+    # def getheight(self,value):
+    #     self.search(self.head,value)
+
+    # def isroot(self,node):
+    #     if node.data==self.head:
+    #         return True
+    #     else:
+    #         return False
+
+    def rotateleft(self, rotroot):
```

Comment:
The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpc8nsbcsj/avl_tree/avl.py
+++ b/tmp/tmpc8nsbcsj/avl_tree/avl.py
@@ -103,7 +103,7 @@
 
     def printme(self, root):
         if root is not None:
-            print (root.data)
+            print(root.data)
             self.printme(root.left)
             self.printme(root.right)
 
```